sdk/core/System.ClientModel/api/System.ClientModel.net6.0.cs

```diff
@@ -25,14 +25,12 @@ public partial class ClientResult
         public static System.ClientModel.ClientResult<T> FromValue<T>(T value, System.ClientModel.Primitives.PipelineResponse response) { throw null; }
         public System.ClientModel.Primitives.PipelineResponse GetRawResponse() { throw null; }
     }
-    public partial class ClientResultException : System.Exception, System.Runtime.Serialization.ISerializable
+    public partial class ClientResultException : System.Exception
```

Comment:
Confirmed with @GrabYourPitchforks that it would not be a breaking change to add this back to the type after the initial GA.

@GrabYourPitchforks shared the following:

> If v1 of a type does not implement `ISerializable`, then it is a breaking change for v2 of that type to implement `ISerializable` only if all of the following hold:
> 1. v1 of the type was not sealed; _and_
> 1. v1 of the type had the `[Serializable]` attribute applied to it.
> So this means that if v1 is sealed and v2 is unsealed, and as part of this you also introduce `ISerializable`, this is not a breaking change.

Since this PR removes the `[Serializable]` attribute from `ClientResultException`, it will not be a breaking change to add the implementation of `ISerializable` with the serialization construct when it's needed for the Azure.Core 2.0 integration.  

We could also seal the type to be extra safe and unseal it when needed for Azure.Core 2.0 integration -- let me know if that's of interest to reviewers of this PR.